consensus/src/quorum_store/utils.rs

```diff
@@ -249,19 +253,30 @@ impl ProofQueue {
     }
 
     fn remaining_txns_without_duplicates(&self) -> u64 {
+        // txn_summary_num_occurrences counts all the unexpired and uncommitted proofs that have txn summaries
+        // in batch_summaries.
         let mut remaining_txns = self.txn_summary_num_occurrences.len() as u64;
 
-        // If a batch_key is not in batches_with_txn_summary, it means we've received the proof but haven't receive the
-        // transaction summary of the batch from batch coordinator. Add the number of txns in the batch to remaining_txns.
+        // For the unexpired and uncommitted proofs that don't have transaction summaries in batch_summaries,
+        // we need to add the proof.num_txns() to the remaining_txns.
         remaining_txns += self
-            .batch_to_proof
-            .iter()
-            .filter_map(|(batch_key, proof)| {
-                if proof.is_some() && !self.batch_to_txn_summaries.contains_key(batch_key) {
-                    Some(proof.as_ref().unwrap().0.num_txns())
-                } else {
-                    None
-                }
+            .author_to_batches
+            .values()
+            .map(|batches| {
+                batches
+                    .keys()
+                    .map(|batch_sort_key| {
+                        if !self.batch_summaries.contains_key(&batch_sort_key.batch_key) {
+                            if let Some(Some((proof, _))) =
+                                self.batch_to_proof.get(&batch_sort_key.batch_key)
+                            {
+                                // The batch is included in author_to_batches, batch_to_proof, but not in batch_summaries
```

Comment:
This will create the duplication issue - can you explain when this happens and anything we can do to resolve it?

Reply:
Creating a proof needs signatures from the quorum. The other validators may not have the batch available locally. So, it's possible that a validator doesn't receive the batch summary, but receives the proof for the batch. In that case, we have to add proof.num_txns() to the calculation without any deduplication. 

If the validator receives the batch summary after receiving the proof, and then calculate the remaining txns, that's still fine. The code will do deduplication when calculating remaining txns.